Combine My Subscriptions and Marketplace back into one Extensions page

andfinally opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

#### Closes WCCOM issues

https://github.com/Automattic/woocommerce.com/issues/11663
https://github.com/Automattic/woocommerce.com/issues/11665

#### Changes

Reverts https://github.com/woocommerce/woocommerce/pull/30380, which split out the content of the `WooCommerce.com Subscriptions` tab on the main Extensions page (i.e. `wp-admin/admin.php?page=wc-addons&section=helper`) to a new separate page `My Subscriptions`. This meant that fewer people were seeing the extensions listings.

In this PR, we're restoring the earlier arrangent, with `My Subscriptions` being reach via a tab on the `Extensions` page. We're also making the extensions category list more obvious on desktop – with categories hidden in a dropdown, people were less aware of the different types of extensions available.

#### Related WC Admin PRs which will be included in WC 6.0

https://github.com/woocommerce/woocommerce-admin/pull/7901 – sets the page title at the top of both pages back to "Extensions".
https://github.com/woocommerce/woocommerce-admin/pull/7902 – reverts to the previous menu arrangement in the new style nav.

### How to test the changes in this Pull Request:

1. Check out this branch on your local environment.
2. Go to WP Admin and observe the My Subscriptons and Marketplace items have been replaced with an Extensions menu item. If any of your WooCommerce extensions happen to need updating, the menu item should show an update count bubble – please see the screenshot for example. Otherwise you won't see the bubble.
3. View the Extensions page.  (The My Subscriptions tab will also show an update count bubble, if any extensions are due for an update. But if not, you won't see one.)
4. Make sure you can navigate to the My Subscriptions page and back again.
5. Try the links under `Browse Categories`. They should show different lists of extensions.
6. Try the page on a mobile device. 
  i. It should look like the screenshot at bottom.
  ii. The My Subscriptions link should take you to the subscriptions page.
  iii. The Browse Extensions tab on that page should return you to the WooCommerce Marketplace page.
  iv. The Browse Categories dropdown list should continue to work.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

Update - Merge Marketplace and My Subscriptions pages back into one Extensions page.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### Screenshots

#### Before

<img width="500" src="https://user-images.githubusercontent.com/1647564/141128534-5d7d0190-8f4f-4c15-b862-e8c911f47b02.png">

#### After

<img width="500" src="https://user-images.githubusercontent.com/1647564/141116962-cca0db70-3cc8-49bf-97a3-2e4bcdffeeee.png">

#### Mobile

<img width="350" src="https://user-images.githubusercontent.com/1647564/142998135-9373d354-451b-4647-ab68-312727e373ab.png">

andfinally committed:
Reverting changes in https://github.com/woocommerce/woocommerce/pull/30380, preparatory to merging the My Subscriptions and Marketplace pages back into one Extensions page.

andfinally committed:
Moved category nav to a separate file.

andfinally committed:
Added main nav tabs.

andfinally committed:
Added category nav, styled the mobile version.

andfinally committed:
Experiments with doing category tabs as a list.

andfinally committed:
Styling category menu as a list of links in rectangular pills on desktop.

andfinally committed:
Adjusting margins of product list in in-app marketplace.

andfinally committed:
Removed `submenu_file` filter which called deleted `update_menu_highlight` method.

andfinally committed:
Restored styling for update count bubble on My Subscriptions tab.

andfinally committed:
Styling update bubble on My Subscriptions tab.

andfinally committed:
Addressing design feedback. Restyled category pills to make them more obvious. Added `Browse categories` title before category menu on desktop. Changed `.addon-product-group-title` to an h2.

jeffstieler commented on plugins/woocommerce/includes/admin/class-wc-admin-menus.php:
```diff
@@ -36,7 +36,6 @@ public function __construct() {
 		add_filter( 'custom_menu_order', array( $this, 'custom_menu_order' ) );
 		add_filter( 'set-screen-option', array( $this, 'set_screen_option' ), 10, 3 );
-		add_filter( 'submenu_file', array( $this, 'update_menu_highlight' ), 10, 2 );
 		add_filter( 'admin_title', array( $this, 'update_my_subscriptions_title' ) );
```
This needs to be removed.

```
Warning: call_user_func_array() expects parameter 1 to be a valid callback, class 'WC_Admin_Menus' does not have a method 'update_my_subscriptions_title' in /srv/www/wordpress-default/public_html/wp-includes/class-wp-hook.php on line 305
```

andfinally replied:
Ah thanks for spotting this! Removed in 54a542bc575c829134da95e745e2a49be2f4f07d.

jeffstieler commented on plugins/woocommerce/includes/admin/views/html-admin-page-addons.php:
```diff
@@ -39,31 +41,35 @@
 
 	<div class="wrap">
 		<div class="marketplace-content-wrapper">
+			<?php require 'html-admin-page-addons-category-nav.php'; ?>
```
Just noting here that other template files prepend `require` paths with the `__DIR__` magic constant.

andfinally replied:
Thanks, added in 54a542bc575c829134da95e745e2a49be2f4f07d.

jeffstieler reviewed (changes requested):
Functionality tested well, just some minor changes needed in the code.

andfinally committed:
Addressing feedback from https://github.com/woocommerce/woocommerce/pull/31085#pullrequestreview-806319880.

- Removing addition of `admin_title` filter calling deleted `update_my_subscriptions_title` method.
- Prepending `__DIR__` constant before path in `require` call.

jeffstieler reviewed (approved):
LGTM!

andfinally commented:
Thanks a lot Jeff!
